### PR TITLE
Adding missing content about Archiving in docs

### PR DESCRIPTION
**Pull request by da-katmurp**
It seems strange for such a key concept to not have any info in docs so I wasn't surprised to find it in: /daml/intro/2_DamlScript.html. The problem is when key concepts are embedded in pages which are essentially about something completely different, they become impossible to find, don't come up in search, the forum fills up with questions, ..etc etc.

There is another issue with this page - and possibly others. It starts off by telling you what you're going to learn in a bulleted list. We are programmed to expect those bullets as headers on the page, and I see other pages in this section are behaving better on this. When those headers don't appear, it's confusing for the reader. 

**Suggest**: a full rewrite of this section by tech authors, also to make it easier to add and remove content by removing the numbering and pre/post-ambles.

This PR edits the info on https://docs.daml.com/daml/intro/2_DamlScript.html to be more succinct. 
Also, the archive function was missing from the reference so I added it there: /daml/reference/updates.html#archive

Requesting initial review and correctness edit from @leo-da who raised the ticket here: https://trello.com/c/EEET2z11/270-add-info-on-archiving

### CODE REVIEW

**Comment by loiskilberg-da** [comment]
Added a docs ticket for rewriting this page as a separate task: https://trello.com/c/0DpkX6EU/279-improve-writing-on-this-page-daml-intro-2damlscripthtml

**Review by loiskilberg-da — APPROVED** [review]

File: docs/source/daml/reference/updates.rst [hunk 1]
```diff
@@ -44,6 +44,20 @@ do
...
+archive
+*******
+
+.. code-block:: daml
+
+   archiveCmd ContractId
+
+- ``archiveCmd`` function.
+- Archives a contract already created and residing on the ledger. The contract is fetched by its unique contract identifier ``ContractId <name of template>``.
+- Returns a transaction id on success.
```
Comment (leonidr-da) [inline docs/source/daml/reference/updates.rst:56]:
```suggestion
- Returns unit.
```

**Review by leonidr-da — CHANGES_REQUESTED** [review]
Good start, added some clarifications.

File: docs/source/daml/reference/updates.rst [hunk 2]
```diff
@@ -44,6 +44,20 @@ do
 
 .. _daml-ref-create:
 
+archive
```
[the remark this answers is not in the document]
Comment (da-katmurp) [reply docs/source/daml/reference/updates.rst:47]: ok, that's true

**Comment by da-katmurp** [comment]
Accepted all suggestions. Added extra chunk about implicit Archive choice in `updates.rst`.

**Review by leonidr-da — APPROVED** [review]

**Comment by leonidr-da** [comment]
@da-katmurp approved, thank you.

**Pull request merged**